src/plugins/ai_assistant_management/observability/public/hooks/use_create_knowledge_base_entry.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { IHttpFetchError, ResponseErrorBody } from '@kbn/core/public';
+import { i18n } from '@kbn/i18n';
+import type { KnowledgeBaseEntry } from '@kbn/observability-ai-assistant-plugin/common/types';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useAppContext } from '../context/app_context';
+import { REACT_QUERY_KEYS } from '../constants';
+
+type ServerError = IHttpFetchError<ResponseErrorBody>;
+
+export function useCreateKnowledgeBaseEntry() {
+  const {
+    http,
+    notifications: { toasts },
+  } = useAppContext();
+  const queryClient = useQueryClient();
+
+  return useMutation<
+    void,
+    ServerError,
+    {
+      entry: Omit<
+        KnowledgeBaseEntry,
+        '@timestamp' | 'confidence' | 'is_correction' | 'public' | 'labels'
+      >;
+    }
+  >(
+    [REACT_QUERY_KEYS.CREATE_KB_ENTRIES],
+    ({ entry }) => {
+      const body = JSON.stringify(entry);
+      return http.post(`/internal/management/ai_assistant/observability/kb/entries/save`, { body });
```

Comment:
Doesn't the API expect the entry to have all the fields?

Reply:
It does not:

https://github.com/elastic/kibana/blob/d8e5339e212f5702b64164794ca86f084995ec95/src/plugins/ai_assistant_management/observability/server/routes/knowledge_base/route.ts#L103C5-L103C5